packages/amplify-graphql-api-construct-tests/src/graphql-request.ts

```diff
@@ -1,19 +1,23 @@
-import { default as fetch, Request } from 'node-fetch';
+import { IHttpRequest } from '@smithy/protocol-http';
+import { QueryParameterBag } from '@smithy/types';
+import { makeSignedRequest, makeTemporaryCredentialsProvider, urlToHttpRequestOptions } from './utils';
 
 export type GraphqlResponse = {
   statusCode: number;
   body: any;
 };
 
-export const graphqlRequest = async (apiEndpoint: string, payload: any): Promise<GraphqlResponse> => {
+export const graphqlRequest = async (request: Request): Promise<GraphqlResponse> => {
```

Comment:
I toyed with the idea of making a totally standalone sigv4-centric request stack, but opted to refactor the existing call utils primarily because of the error handling below. Since we override the status code if there are any errors, which is not AppSync's normal behavior, I didn't want to introduce unexpected differences in error handling behavior based on the authorization mode.